src/http/ClientRequest/ClientRequestOverride.ts

```diff
@@ -218,13 +218,13 @@ export function createClientRequestOverrideClass(
         // it always performs respecting this `ClientRequest` restoration.
         originalClientRequest = null as any
 
-        req = performOriginalRequest(options, callback)
+        req = performOriginalRequest(options)
 
         debug('re-applying patches...')
         http.ClientRequest = ClientRequest
         originalClientRequest = ClientRequest
       } else {
-        req = performOriginalRequest(options, callback)
+        req = performOriginalRequest(options)
```

Comment:
So the callback will still get called, because we've listened to the `response` event previously?

https://github.com/mswjs/node-request-interceptor/blob/2e0c204eaca6e8deebce302858ed2edb02afb1be/src/http/ClientRequest/ClientRequestOverride.ts#L75-L77

This `req` instance is the unpatched original `ClientRequest` instance. Have you figured it out how come it inherits the callback that we've assigned to the patched (overridden) instance?

Reply:
I've got it: later, with the original request, we define:

```js
req.on('response', (response) => {
  debug(response.statusCode, options.method, url.href)
  this.emit('response', response)
})
```

This triggers the callback we passed to the patched request: `this.on('response', callback)`.

Reply:
Yeah, that's the place! Sorry for the delayed reaction, comment notification drowned in others and I completely missed it 😓